rosserial_arduino/src/ros_lib/ArduinoHardware.h

```diff
@@ -57,14 +57,14 @@
 #elif (defined(__STM32F1__) and !(defined(USE_STM32_HW_SERIAL))) or defined(SPARK) 
   // Stm32duino Maple mini USB Serial Port
   #define SERIAL_CLASS USBSerial
-#else 
-  #include <HardwareSerial.h>  // Arduino AVR
-  #if defined(USE_USBCDC)
-    // ESP32-S2 ESP32-C3
-    #define SERIAL_CLASS USBCDC
-  #elif defined(USE_HWCDC)
-    // ESP32-S3
-    #define SERIAL_CLASS HWCDC
+#else
+  #if ARDUINO_USB_CDC_ON_BOOT 
+    #include <HardwareSerial.h>  // Arduino AVR
+    #if ARDUINO_USB_MODE 
+      #define SERIAL_CLASS HWCDC
+    #else
+      #define SERIAL_CLASS USBCDC
+    #endif
```

Comment:
```suggestion
  #include <HardwareSerial.h>  // Arduino AVR
  #define SERIAL_CLASS decltype(Serial)
```

should work in any combination of preprocessor flags `ARDUINO_USB_CDC_ON_BOOT` and `ARDUINO_USB_MODE` without a change in behavior. The compiled code would be exactly the same. Just without the need to replicate the conditions in [`HardwareSerial.h`](https://github.com/espressif/arduino-esp32/blob/cf448906b3836fbe9368934713b697469254c62f/cores/esp32/HardwareSerial.h#L346-L353).

Calling `iostream->begin(baud_)` in `init()` or not is a different story. From the code in [`HWCDC::begin()`](https://github.com/espressif/arduino-esp32/blob/cf448906b3836fbe9368934713b697469254c62f/cores/esp32/HWCDC.cpp#L244-L246) I would still think that the USB host has to enumerate the device twice on reboot (disconnect - connect - disconnect - connect cycle), if `begin()` was called twice, once from `arduino-esp32`'s main function and once in `ArduinoHardware::init()`. But usually the ESP32 controller is already booted long before the PC and it _should_ never reboot, so it does not matter much.